services/src/main/java/org/exoplatform/officeonline/OfficeOnlineEditorService.java

```diff
@@ -91,6 +142,268 @@ public void start() {
     LOG.debug("PowerPoint view URL: " + powerPointView);
   }
 
+  /**
+   * Check file info.
+   *
+   * @param userSchema the user schema
+   * @param userHost the user host
+   * @param userPort the user port
+   * @param workspace the workspace
+   * @param fileId the file id
+   * @param accessToken the access token
+   * @return the map
+   * @throws RepositoryException the repository exception
+   */
+  public Map<String, Serializable> checkFileInfo(String userSchema,
```

Comment:
This looks as pure WOPI stuff as well as other methods in this commit. Let's move such API calls out of the service to WOPI-related object (its name good to set to 'server', 'provider' or 'SPI').